packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/SystemServicesHostApiImpl.java

```diff
@@ -108,4 +120,16 @@ public void stopListeningForDeviceOrientationChange() {
       deviceOrientationManager.stop();
     }
   }
+
+  /** Returns a path to be used to create a temp file in the current cache directory. */
+  @Override
+  @NonNull
+  public String getTempFilePath(@NonNull String prefix, @NonNull String suffix) {
+    try {
+      File path = File.createTempFile(prefix, suffix, context.getCacheDir());
+      return path.toString();
+    } catch (IOException | SecurityException e) {
+      throw new RuntimeException(e);
```

Comment:
Should this be a typed exception?

Reply:
At the time it was written, it was only possible to use RuntimeException here (the general pattern was that this method is wrapped in a try catch anyways in GeneratedCameraXLibrary, and then GeneratedCameraXLibrary would throw a PlatformException).

Since updating to pigeon 9.10, we can throw FlutterError instead (there is some discussion about the old pattern vs new FlutterError [on this old PR](https://github.com/flutter/packages/pull/3561#event-8858727980)). I will change here to throw a FlutterError.